From bf42dba499e053196b9235aae9987f395c4a4b7e Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 29 May 2012 20:44:18 +0100
Subject: [PATCH 2/4] configure: Allow user defined C compiler flags.

---
 configure.ac | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure.ac b/configure.ac
index 83455a3b6..213392b25 100644
--- a/configure.ac
+++ b/configure.ac
@@ -638,6 +638,10 @@ AS_CASE([$host],
       internal_cflags="$cc_warnings"],
     [common_cflags="-O"])])
 
+# Allow CFLAGS and LDFLAGS to be added.
+common_cflags="$common_cflags $CFLAGS"
+cclibs="$cclibs $LDFLAGS"
+
 internal_cppflags="-DCAML_NAME_SPACE $internal_cppflags"
 
 # Enable SSE2 on x86 mingw to avoid using 80-bit registers.
-- 
2.32.0

